packages/edit-site/src/components/left-sidebar/navigation-panel/template-navigation-items.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	__experimentalNavigationItem as NavigationItem,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import TemplatePreview from './template-preview';
+import { NavigationPanelPreviewFill } from '.';
+import { ITEM_CONTENTS } from './constants';
+
+export default function TemplateNavigationItems( {
+	entityType = 'wp_template',
+	templates,
+	onActivate,
+} ) {
+	const [ hoveredTemplate, setHoveredTemplate ] = useState();
+
+	const onMouseEnterTemplate = ( template ) => setHoveredTemplate( template );
+	const onMouseLeaveTemplate = () => setHoveredTemplate( null );
+
+	if ( ! templates ) {
+		return null;
+	}
+
+	if ( ! Array.isArray( templates ) ) {
+		templates = [ templates ];
+	}
+
+	const itemPrefix =
+		entityType === 'wp_template_part' ? 'template-part' : 'template';
+
+	return (
+		<>
+			{ templates.map( ( template ) => {
+				const { title, description } =
+					ITEM_CONTENTS[ template.slug ] ?? {};
+				const key = `${ itemPrefix }-${ template.id }`;
+
+				return (
+					<NavigationItem
+						className="edit-site-navigation-panel__template-item"
+						key={ key }
+						item={ key }
+						title={ title || template.slug }
+					>
+						<Button
+							onClick={ () => onActivate( template.id ) }
+							onMouseEnter={ () =>
+								onMouseEnterTemplate( template )
+							}
+							onMouseLeave={ onMouseLeaveTemplate }
+						>
+							{ title || template.slug }
+							{ description && (
+								<div className="edit-site-navigation-panel__template-item-description">
+									{ description }
+								</div>
```

Comment:
As [mentioned in another issue](https://github.com/WordPress/gutenberg/issues/25755#issuecomment-702063762), templates _should/could_ allow for user-defined descriptions in the `excerpt` field.

I guess we could do the same we do for the title/slug, and add `template.excerpt` as a fallback — or, as I mentioned in another comment, the "main" field, and treat `description` as fallback.